php-frontend/src/main/java/org/sonar/php/symbols/MethodSymbolImpl.java

```diff
@@ -44,17 +50,90 @@ public String name() {
 
   @Override
   public Trilean isOverriding() {
-    Optional<ClassSymbol> superClass = owner.superClass();
-    while (superClass.isPresent()) {
-      if (superClass.get().isUnknownSymbol()) {
+    if (isOverriding == null) {
+      isOverriding = checkSuperClassesAndInterfacesForDeclaration();
+    }
+    return isOverriding;
+  }
+
+  /**
+   * Check all interfaces of a class before searching in the super class
+   * to validate if a method has been declared in an implemented interface or extended super class
+   */
+  private Trilean checkSuperClassesAndInterfacesForDeclaration() {
```

Comment:
This algorithm seems quite complex to me. 
I think that we can consider superclasses and interfaces in the same way. 
I would implement a single non-recursive method which iterates over all the ClassSymbols of the hierarchy with a kind of "worklist".